### ISSUE TITLE
Deletion of files doesn't trigger events

### ISSUE THREAD

**Issue by LvanderGoten** [eradman/entr#88]
Hi,
thanks a lot for your great project, I think that entr is a really useful tool.

I want do output "Hello" whenever a directory or (some other descendant subdirectory) changes. The following works in the case of **adding** files, but it doesn't work when I delete a file (**no** event is triggered).

```
while true; do
    sleep 2s;
    find /home/user/Code -name '*' -type d | entr -dp echo "Hello";
done
```

Another approach I tried is to remove `-type d` from `find` to incorporate the files as well:
```
while true; do
    sleep 2s;
    find /home/user/Code -name '*' | entr -dp echo "Hello";
done
```

When I delete a file `testfile` now I get the following message and **no** event is triggered:
```
 entr: cannot open '/home/user/Code/some_subdir/testfile': No such file or directory
```

I have two questions now:

1. *Is it possible to trigger events when files get deleted?*
2. *If I'm removing `sleep 2s`  it is no longer possible to quit the program via `CTRL-C`. Is there something I'm missing (`-r` doesn't seem to work)*

**Comment by eradman** [comment]
The first problem is a bug on Linux.  I'm very surprised that nobody noticed this before!

`entr` was monitoring for files that are created in directories, but not files that are deleted

```c
#define IN_CREATE         0x00000100        /* Subfile was created.  */
#define IN_DELETE         0x00000200        /* Subfile was deleted.  */
```

Fixed in commit 90af966

The sleep statement is required for `Ctrl-C` because in a while loop because shell is rapidly transferring control of the terminal so SIGINT is sent to the foreground process and not the shell. A slightly shorter way to type it is. One workaround is to look for an exit status of `0` and use that to exit the loop

```
while true; do
    find /home/user/Code | entr -dp echo "Hello" && break
done

**Comment by LvanderGoten** [comment]
Thanks for the great answer!
